### PR TITLE
Add ability to identify short-form `StreamInfoItem`s

### PR DESCRIPTION
**Pull request by chowder**
Addresses https://github.com/TeamNewPipe/NewPipeExtractor/issues/942 

- [X] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [X] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [X] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

### CODE REVIEW

**Comment by TobiGr** [comment]
thank you. I think providing the same information for `StreamInfo` would be nice. Can you do this as well?

**Comment by chowder** [comment]
I'll have a look to see if I can do the same for `StreamInfo`

Also, I noticed that a number of tests are failing in the CI action, but these failures are unrelated to my changes (the test that I introduced is passing), do I have to fix them?

**Comment by TobiGr** [comment]
You don't need to fix the unrelated tests, I created a PR to fix them.
However, [this test](https://github.com/TeamNewPipe/NewPipeExtractor/actions/runs/3276083975/jobs/5393324731#step:5:246) should be fixed by you.

**Comment by chowder** [comment]
Ah thanks, I've regenerated the test mocks now. 

I've also added the same `isShortFormContent` methods for `StreamInfo` and `StreamExtractor`, but haven't added any actual implementations; for YouTube in particular I couldn't find any fields that distinguish between regular and short videos on the `/watch?v=` links (using [this video](https://www.youtube.com/watch?v=R1Ehb3JA-cM) as reference.

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamInfoItemExtractor.java [hunk 1]
```diff
@@ -324,4 +324,16 @@ public String getShortDescription() throws ParsingException {
...
+
+    @Override
+    public boolean isShortFormContent() throws ParsingException {
+        try {
+            final String webPageType = videoInfo.getObject("navigationEndpoint")
+                    .getObject("commandMetadata").getObject("webCommandMetadata")
+                    .getString("webPageType");
+            return !isNullOrEmpty(webPageType) && webPageType.equals("WEB_PAGE_TYPE_SHORTS");
+        } catch (final Exception e) {
+            throw new ParsingException("Could not determine if short-form content", e);
```
Comment (AudricV) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamInfoItemExtractor.java:336]: This seems not understandable as an exception message: I suggest using something like this (given by @opusforlife2 to me):

```suggestion
            throw new ParsingException("Could not determine if this is short-form content", e);
```

File: extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java [hunk 2]
```diff
@@ -554,6 +554,16 @@ public List<MetaInfo> getMetaInfo() throws ParsingException {
         return Collections.emptyList();
     }
 
+    /**
+     * Whether the stream is a short-form content. These are content in the style of TikTok,
+     * YouTube Shorts, or Instagram Reels videos.
```
Comment (AudricV) [inline extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java:559]:
```suggestion
     * Whether the stream is a short-form content.
     *
     * <p>
     * Short-form contents are contents in the style of TikTok, YouTube Shorts, or Instagram Reels videos.
     * </p>
```

File: extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamInfoItem.java [hunk 3]
```diff
@@ -130,6 +131,15 @@ public void setUploaderVerified(final boolean uploaderVerified) {
...
 
+    public boolean isShortFormContent() {
+        return shortFormContent;
+    }
+
+    public void setShortFormContent(final boolean shortFormContent) {
+        this.shortFormContent = shortFormContent;
+    }
+
+
```
Comment (AudricV) [inline extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamInfoItem.java:142]: Remove this extra blank line.

File: extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamInfoItemExtractor.java [hunk 4]
```diff
@@ -127,4 +127,14 @@ public interface StreamInfoItemExtractor extends InfoItemExtractor {
     default String getShortDescription() throws ParsingException {
         return null;
     }
+
+    /**
+     * Check if the stream is a short-form content
+     *
+     * @return {@code true} if the stream is a short-form content
+     * @throws ParsingException thrown if there is an error in the extraction
+     */
```
Comment (AudricV) [inline extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamInfoItemExtractor.java:136]: You should use the first description everywhere (with my changes, I think), which give us more details about what is this type of contents.

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/search/YoutubeSearchExtractorTest.java [hunk 5]
```diff
@@ -379,4 +380,49 @@ public void testVideoDescription() throws IOException, ExtractionException {
...
+            final List<InfoItem> items = extractor.getInitialPage().getItems();
+            boolean hasShortFormContent = false;
+            for (InfoItem item : items) {
+                if (((StreamInfoItem) item).isShortFormContent()) {
+                    hasShortFormContent = true;
+                    break;
+                }
+            }
+
+            assertTrue(hasShortFormContent);
```
Comment (AudricV) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/search/YoutubeSearchExtractorTest.java:425]: Remove the unneeded `public` test method modifier here. Also, I recommend using for the test Java 8's `Stream` API instead:

```suggestion
        void testShortFormContent() throws IOException, ExtractionException {
            assertTrue(extractor.getInitialPage()
                    .getItems()
                    .stream()
                    .filter(StreamInfoItem.class::isInstance)
                    .map(StreamInfoItem.class::cast)
                    .anyMatch(StreamInfoItem::isShortFormContent));
```

**Review by AudricV — CHANGES_REQUESTED** [review]
Thank you very much! Here are some changes you need to do in this PR.

Could you also rebase it?

**Comment by FireMasterK** [comment]
I've addressed the comments and rebased this PR at https://github.com/FireMasterK/NewPipeExtractor/tree/shorts

**Comment by chowder** [comment]
Thank you so much @FireMasterK, I've cherry-picked your commit addressing the remaining comments.

**Review by FireMasterK — APPROVED** [review]

**Comment by chowder** [comment]
> Except that I am not sure if we should use setIsShortFormContent(boolean) instead of setShortFormContent(boolean) for setters

I agree, but I was trying to be consistent with other boolean setters - e.g. [`isUploaderVerified`/`setUploaderVerified`](https://github.com/TeamNewPipe/NewPipeExtractor/blob/b1a899fd47ca5db9bc7cf65ac70a5f20178fb47d/extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamInfo.java#L537) 😅

**Comment by FireMasterK** [comment]
Intellij's setter generator would generate them as `setShortFormContent(boolean)`, I would keep it the way it is. (and to keep things consistent)

**Pull request merged**